apis/model/v1alpha1/model_types.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2020 The Alibaba Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ModelSpec defines the desired state of Model
+// Model defines the storage where the model is located, the imageRepo for pushing the generated model container.
+// Each model has a series of ModelVersion associated with it.
+type ModelSpec struct {
+	// Storage is the location where the model is stored.
+	// +optional
+	Storage *Storage `json:"storage,omitempty"`
+
+	// ImageRepo is the image repository to push the generated image. e.g. docker hub.  "kubernetes/pause"
+	// +required
+	ImageRepo string `json:"imageRepo,omitempty"`
+}
+
+type Storage struct {
+	AliCloudNas  *AliCloudNas  `json:"aliCloudNas,omitempty"`
+	LocalStorage *LocalStorage `json:"localStorage,omitempty"`
+}
+
+// If the localStorage is defined in parent model, nodeName must be set, that is, all output models will be stored on the same node.
```

Comment:
Is it necessary to enforce all the modelVersions to be stored in the same node?

Reply:
If it is specified in parent model's storage, then we assume it is on the same node, because at that time, we don't know which node the model is stored.
It can on any nodes for ModelVersion.  After the job finished and generated the model, we know which node stores the model and we can create a ModelVersion with the corresponding node.

Reply:
I added a comment to explain.